src/server/string_family.cc

```diff
@@ -536,16 +536,31 @@ struct MGetResponse {
 // fetch_mask values
 constexpr uint8_t FETCH_MCFLAG = 0x1;
 constexpr uint8_t FETCH_MCVER = 0x2;
-MGetResponse OpMGet(fb2::BlockingCounter wait_bc, uint8_t fetch_mask, const Transaction* t,
-                    EngineShard* shard) {
+
+template <typename Iter> using SearchKey = std::function<OpResult<Iter>(string_view)>;
+
+// A find operation which can mutate, for commands which can write, eg GAT
+using SearchMut = SearchKey<DbSlice::Iterator>;
+
+// Const find operation, for read-only commands, eg MGet
+using SearchConst = SearchKey<DbSlice::ConstIterator>;
+
+template <typename Iter>
+MGetResponse CollectKeys(BlockingCounter wait_bc, uint8_t fetch_mask, const Transaction* t,
+                         EngineShard* shard, SearchKey<Iter> find_op) {
   ShardArgs keys = t->GetShardArgs(shard->shard_id());
   DCHECK(!keys.Empty());
 
   auto& db_slice = t->GetDbSlice(shard->shard_id());
```

Comment:
move `db_slice` to the block where you use it

Reply:
moved